pyintegration/deephaven2/stream/kafka/producer.py

```diff
@@ -0,0 +1,183 @@
+        schema_namespace (str): when 'publish_schema' is True, the namespace for the generated schema to be registered in
+            Schema Registry Server.
+        column_properties (Dict[str, str]): when 'publish_schema' is True, specifies the properties of the columns
+            implying particular Avro type mappings for them. In particular, column X of BigDecimal type should specify
+            properties 'x.precision' and 'x.scale'.
+
+    Returns:
+        a KeyValueSpec
+
+    Raises:
+        DHError
+    """
+    try:
+        field_to_col_mapping = dtypes.HashMap(field_to_col_mapping)
+        column_properties = dtypes.Properties(column_properties)
+        include_only_columns = dtypes.HashSet(include_only_columns)
+        include_only_columns = _JKafkaTools.predicateFromSet(include_only_columns)
+        exclude_columns = dtypes.HashSet(exclude_columns)
+        exclude_columns = _JKafkaTools.predicateFromSet(exclude_columns)
+
+        return KeyValueSpec(_JKafkaTools_Produce.avroSpec(schema, schema_version, field_to_col_mapping, timestamp_field,
+                                                          include_only_columns, exclude_columns, publish_schema,
+                                                          schema_namespace, column_properties))
+    except Exception as e:
+        raise DHError(e, "failed to create a Kafka key/value spec.") from e
+
+
+def json_spec(include_columns: List[str] = None, exclude_columns: List[str] = None, mapping: Dict[str, str] = None,
+         nested_delim: str = None, output_nulls: bool = False, timestamp_field: str = None) -> KeyValueSpec:
+    """ Creates a spec for how to generate JSON data when producing a Kafka stream from a Deephaven table.
+
+    Args:
+        include_columns (List[str]): the list of Deephaven column names to include in the JSON output as fields,
+            default is None, meaning all except the ones mentioned in the 'exclude_columns' argument . If not None,
+            the 'exclude_columns' must be None.
+        exclude_columns (List[str]): the list of Deephaven column names to omit in the JSON output as fields, default
+            is None, meaning no column is omitted. If not None, include_columns must be None.
+        mapping (Dict[str, str]): a mapping from column names to JSON field names.  Any column name implied by earlier
+            arguments and not included as a key in the map implies a field of the same name. default is None,
+            meaning all columns will be mapped to JSON fields of the same name.
+        nested_delim: if nested JSON fields are desired, the field separator that is used for the field names
+            parameter, or None for no nesting (default). For instance, if a particular column should be mapped
+            to JSON field X nested inside field Y, the corresponding field name value for the column key
+            in the mapping dict can be the string "X.Y", in which case the value for nested_delim should be "."
```

Comment:
I can figure out the nesting stuff, I think, but a paragraph before args could be helpful.